yarn-project/archiver/src/archiver/kv_archiver_store/log_store.ts

```diff
@@ -44,21 +52,50 @@ export class LogStore {
   addLogs(blocks: L2Block[]): Promise<boolean> {
     return this.db.transaction(() => {
       blocks.forEach(block => {
-        void this.#noteEncryptedLogs.set(block.number, block.body.noteEncryptedLogs.toBuffer());
-        void this.#encryptedLogs.set(block.number, block.body.encryptedLogs.toBuffer());
-        void this.#unencryptedLogs.set(block.number, block.body.unencryptedLogs.toBuffer());
+        void this.#noteEncryptedLogsByBlock.set(block.number, block.body.noteEncryptedLogs.toBuffer());
+        block.body.noteEncryptedLogs.txLogs.forEach(txLogs => {
+          const noteLogs = txLogs.unrollLogs();
+          noteLogs.forEach(noteLog => {
+            const tag = new Fr(noteLog.data.subarray(0, 32));
+            const hexHash = noteLog.hash().toString('hex');
+            // AztecMultiMap doesn't handle storing buffers well. Using the body 'ordered-binary' encoding returns an error
+            // when trying to decode buffers (The number <> cannot be converted to a BigInt because it is not an integer)
+            // This unfortunately leads to some duplication, but using the hash of the note log should be fine
+            // even if we had duplicated tags.
```

Comment:
```suggestion
            // Ideally we'd store all of the logs for a matching tag in an AztecMultiMap, but this type doesn't doesn't
            // handle storing buffers well. The 'ordered-binary' encoding returns an error trying to decode buffers
            // ('the number <> cannot be converted to a BigInt because it is not an integer'). We therefore store
            // instead the hashes of the logs.
```